libraries/AP_SensorHead/AP_SensorHead.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "Protocol.h"
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Baro/AP_Baro.h>
+#include <AP_InertialSensor/AP_InertialSensor.h>
+#include <AP_HAL/AP_HAL.h>
+
+using namespace SensorHead;
+
+/*
+ * Base Class for a Message Handler
+ */
+template <class T>
+class AP_SensorHead_Handler {
+public:
+    virtual void handle(typename T::data_t *data) = 0;
+};
+
+class AP_SensorHead {
+public:
+
+    AP_SensorHead &baro(AP_Baro *baro) {
```

Comment:
these look like registration functions (giving the sensorhead library access to another library). Maybe use the word "register" in the function name? Something like register_baro() ?